src/source/video_source.js

```diff
@@ -166,7 +181,12 @@ class VideoSource extends ImageSource {
     }
 
     hasTransition() {
-        return this.video && !this.video.paused;
+        const result = this.video && (!this.video.paused || this.needsUpdate);
+        if (this.needsUpdate && this.textureDrawn) {
```

Comment:
This check seems superfluous. If these variables are already `false`, setting them to `false` won't change anything and if they're `true` then they get set to `false` anyway.

Reply:
In fact, we could just do

```
this.needsUpdate = false;
this.textureDrawn = false;
return this.video && (!this.video.paused || this.needsUpdate);
```

Reply:
Hey Ryan, in the first update I made to the code i just added what's now the needsUpdate variable and it worked most of the time, but i noticed that occasionally the video wouldn't be rendered after skipping. I assumed that this was due to a race condition between the prepare and hasTransition functions where the hasTransition function would be called after the video was seeked but before the prepare function. This would set the needsUpdate variable back to false before reaching the prepare function so it wouldn't draw the texture for the updated position of the video.  The textureDrawn variable was added to prevent this race condition so I could guarantee that the updated texture has been drawn before the hasTransition function is called so the painter has an updated texture to draw. Once that's complete, the needsUpdate and textureDrawn variables can be set back to false so the painting can resume normally

Reply:
Hey @dwilson2547 did you notice this in a specific environment or anything like that? I've been playing with the code and I see it working (at least in Firefox) as expected with no changes to `hasTransition` at all. I've tried quite a few times and haven't been able to produce a race condition where the video fails to update when skipping forward. Is there a specific way to produce the bug?

`needsUpdate` is necessary in lines 92 and 157, but I can't see why it's necessary in `hasTransition` or why `textureDrawn` is necessary at all.

Reply:
@ryanhamley I haven't been able to purposely reproduce the bug, I actually didn't notice it in my first few days of testing. For me it happened about once in every 50 tries, I can't definitively prove that it was a race condition where hasTransition was run before the prepare function, but that was the only thing that made sense to me. I did my testing in chrome and firefox and I saw the bug 3 times. I checked the console for errors and never saw any so I'm fairly certain it wasn't just a slow buffering issue